DataStructures/Source/LinkedList/LinkedList.cs

```diff
@@ -47,41 +47,42 @@ public void AddAt(int index, T item)
                 return;
             }
 
-            var previous = ElementAt(index - 1, _head);
+            var previous = FindNodeAt(index - 1);
             previous.Next = new Node<T>(item, previous.Next);
             Length++;
         }
 
         public void Remove(T item)
         {
-            var previous = _head;
+            Node<T> previous = null;
+            var node = _head;
             for (int i = 0; i < Length; i++)
```

Comment:
for - should be a bit faster since no additional classes are generated. We know Length so everything is known to iterate over nodes;

foreach - will generate an additional code to iterate over nodes + the source for foreach is unknown